ludwig/models/llm.py

```diff
@@ -197,13 +188,32 @@ def __init__(
     def create_feature_dict(self) -> LudwigFeatureDict:
         return DictWrapper(LudwigFeatureDict())
 
-    def set_generation_config(self, generation_config_dict):
+    @contextlib.contextmanager
+    def use_generation_config(self, generation_config_dict: Optional[Dict[str, Any]] = None):
         """Sets the generation config for the model."""
-        self.generation = GenerationConfig(**generation_config_dict)
-
-    def reset_generation_config(self):
-        """Sets the generation config for th."""
-        self.generation = self.original_generation_config
+        # Save the original generation config so that we can reset it if/when we change it when self.generation gets is
+        # dynamically mutated during 1-off predict calls after fine-tuning.
+        original_generation = copy.deepcopy(self.generation)
+        original_max_new_tokens = self.max_new_tokens
```

Comment:
If we rely on `_set_generation_config(original_generation)`, then we can remove these two LOC.